src/App.js

```diff
@@ -1,50 +1,135 @@
+import Quiz from "./components/Quiz.js";
+import QuizComplete from "./components/QuizComplete.js";
+import QuizResults from "./components/QuizResults.js";
+import questionsData from "./data/questions.json";
+import weightsData from "./data/weights.json";
+import "./App.css";
+import "./styles/styles.css";
+import { getTrackInfo, getToken } from "./api/spotifyApi.js";
+import { fetchYouTubeData } from "./api/youtubeApi.js";
 
 function App() {
   const [currentQuestionIndex, setCurrentQuestionIndex] = useState(-1);
   const [questions, setQuestions] = useState([]);
   const [weights, setWeights] = useState([]);
   const [isQuizComplete, setIsQuizComplete] = useState(false);
+  const [isCalculatedResults, setCalculatedResults] = useState(false);
   const [error, setError] = useState(false);
+  const [dominantGenre, setDominantGenre] = useState(null);
+  const [spotifyLink, setSpotifyLink] = useState('');
+  const [youTubeVideos, setYouTubeVideos] = useState([]);
+
   useEffect(() => {
-    if (isQuizComplete){
-      console.log('Implement get Token');
+    if (isQuizComplete) {
+      console.log("Quiz is Complete");
     }
-  }, [isQuizComplete])
+  }, [isQuizComplete]);
+
+  useEffect(() => {
+    if (isCalculatedResults) {
+      console.log("Calculate the Results");
+    }
+  }, [isCalculatedResults]);
+
   useEffect(() => {
     setError(false);
-    try{
+    try {
       setQuestions(questionsData);
-      setWeights(weightsData); 
-    }
-    catch(e){
+      setWeights(weightsData);
+    } catch (e) {
       console.error(e);
       setError(true);
     }
-  }, [])
-  const handleStartQuiz= () => {
+  }, []);
+
+  const handleStartQuiz = () => {
     setCurrentQuestionIndex(currentQuestionIndex + 1);
-    console.log('Start Quiz');
-  }
+    console.log("Start Quiz");
+  };
+
+  const handleCalculateResults = () => {
+    setCalculatedResults(true);
+    console.log("Calculate the Results!");
+    const dominant = calculateDominantGenre(weights);
+    setDominantGenre(dominant);
+  };
+
+  const resetQuiz = () => {
+    setCurrentQuestionIndex(-1);
+    setIsQuizComplete(false);
+    setCalculatedResults(false);
+    setSpotifyLink('');
+    setYouTubeVideos([]);
+    console.log("Reset Quiz");
+  };
+
+  const calculateDominantGenre = (weights) => {
+    let maxValue = -Infinity;
+    let dominantGenre;
+    for (const genre in weights) {
+      if (weights.hasOwnProperty(genre)) {
+        if (weights[genre] > maxValue) {
+          maxValue = weights[genre];
+          dominantGenre = genre;
+        }
+      }
+    }
+    return dominantGenre;
+  };
+
+  const displaySpotifyLink = async () => {
+    try {
+      const token = await getToken();
+      const trackInfo = await getTrackInfo(token, dominantGenre);
+      setSpotifyLink(trackInfo.tracks[0].external_urls.spotify);
+    } catch(error) {
+      console.error("Error fetching Spotify track:", error);
+    }
+  };
+
+  const fetchYouTubeDataAndDisplay = async () => {
```

Comment:
Some of these functions could probably be broken out into a service or some utils just for general cleanup